### PR TITLE
[Webportal Plugin] Upgrade marketplace plugin

### LINKED ISSUE #2630: Upgrade marketplace plugin
**What would you like to be added**:
Upgrade marketplace plugin.

**Why is this needed**:
Support user to go through protocol jobs in marketplace without submitting job.

**Components that may involve changes**:
* Add a new webportal plugin.
* Abandon outdated marketplace in webportal and rest server.

### PR DESCRIPTION
**Pull request by abuccts**
Close #2630.
Support user to go through protocol jobs in marketplace without submitting job.

### CODE REVIEW

**Comment by abuccts** [comment]
Here're some snapshots:

* List view:
image

* Grid view:
image

File: contrib/marketplace/src/App/MarketplaceList.tsx [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+  listCellDescription: {
+    color: DefaultPalette.white,
+    selectors: {
+      "&:disabled": {
+        color: DefaultPalette.white,
+      },
+    },
+  },
+});
+const cx = classNames.bind(styles);
```
Comment (sunqinzheng) [inline contrib/marketplace/src/App/MarketplaceList.tsx]: I think the variable is not used.

File: contrib/marketplace/src/App/MarketplaceList.tsx [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import React from "react";
+import {
+  DocumentCard, DocumentCardActions, DocumentCardActivity, DocumentCardLogo, DocumentCardStatus, DocumentCardTitle,
+  DefaultPalette, Fabric, Icon, IconButton, Label, Persona, PersonaSize, Stack, Spinner, SpinnerSize, Text, TextField,
+  initializeIcons, mergeStyleSets,
+} from "office-ui-fabric-react";
+import classNames from "classnames/bind";
+import yaml from "yaml";
```
Comment (sunqinzheng) [inline contrib/marketplace/src/App/MarketplaceList.tsx]: nodeca's [js-yaml](https://github.com/nodeca/js-yaml) is more popular.
Have you tried it?

File: contrib/marketplace/src/App/MarketplaceList.tsx (hunk 1 above)
Comment (abuccts) [reply contrib/marketplace/src/App/MarketplaceList.tsx]: yes, removed `classnames` dependence.

File: contrib/marketplace/src/App/MarketplaceList.tsx (hunk 2 above)
Comment (abuccts) [reply contrib/marketplace/src/App/MarketplaceList.tsx]: `js-yaml`'s browser library cannot be used directly in nodejs, so @Gerhut suggested to use `yaml` which is a pure browser library for YAML.

File: contrib/marketplace/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  },
+};
+
+class MarketplacePluginElement extends HTMLElement {
+  public connectedCallback() {
+    const api = this.getAttribute("pai-rest-server-uri") as string;
+    const user = this.getAttribute("pai-user");
+    const token = this.getAttribute("pai-rest-server-token");
+    if (user === null || token === null) {
+      window.location.href = "/login.html";
```
Comment (sunqinzheng) [inline contrib/marketplace/src/index.ts]: `login.html` is deprecated.
User token will be checked in layout.js and anonymous user will be redirected to home page.

File: contrib/marketplace/src/App/MarketplaceList.tsx (hunk 2 above)
Comment (sunqinzheng) [reply contrib/marketplace/src/App/MarketplaceList.tsx]: All right

Comment (abuccts) [reply contrib/marketplace/src/App/MarketplaceList.tsx]: updated

File: contrib/marketplace/src/index.ts (hunk 3 above)
Comment (abuccts) [reply contrib/marketplace/src/index.ts]: removed

**Pull request merged**